apps/barista-backend/src/search/map-content-to-search-result.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  BaPageBuilderContentResult,
+  BaSearchResult,
+  BaSinglePageContent,
+} from '@dynatrace/barista-components/barista-definitions';
+import { Index } from 'lunr';
+
+const EXCERPT_MATCH_LENGTH = 50;
+
+/** Checks if the result has a tag match as well. */
+function isTagMatch(result: Index.Result): boolean {
+  for (const tokenMetadata of Object.values(result.matchData.metadata)) {
+    if (tokenMetadata.tags !== undefined) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * Replace html tags in the content, to mimick the functionality that
+ * cheerio used to produce the index in the first place. Preserve code blocks
+ * to actually match the index positions of the search results.
+ */
+export function replaceHtmlInContent(content: string): string {
+  const reg = /<[^>]*>/gm;
+  let inCodeblock = false;
+  let replacedContent = '';
+  let match = reg.exec(content);
+
+  let lastEndIndex = 0;
+  while (match !== null) {
+    replacedContent = `${replacedContent}${content.slice(
+      lastEndIndex,
+      match.index,
+    )}`;
+    if (match[0].includes('</code')) {
+      inCodeblock = false;
+    }
+    // If we are in a codeblock we need to keep the tags.
+    if (inCodeblock) {
+      lastEndIndex = match.index;
+    } else {
+      lastEndIndex = match.index + match[0].length;
+    }
+
+    if (match[0].includes('<code')) {
+      inCodeblock = true;
+    }
+
+    match = reg.exec(content);
+  }
+
+  // Replace custom HTML entities, to actually match up the length
+  // of the original file.
+  return replacedContent
+    .replace(/&apos;/g, "'")
```

Comment:
multiline flag when the string has carriage return? 

```suggestion
    .replace(/&apos;/gm, "'")
```

Reply:
String does not have carriage returns in the string, as it is not allowed in json format, where we get it from. But it can't hurt.